### PR TITLE
Add TO Client API for Profile Parameters Automation

### PR DESCRIPTION
**Pull request by dmohan001c**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Traffic Control Client 
- Traffic Ops
- CI tests

## What is the best way to verify this PR?
Execute all the Integration tests and make sure the tests passed.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/testing/api/v4/profile_parameters_test.go [hunk 1]
```diff
@@ -144,35 +190,274 @@ func GetTestProfileParameters(t *testing.T) {
...
+	if err != nil {
+		t.Errorf("cannot get Parameter by name '%s': %v - alerts: %+v", firstParameter2.Name, err, paramResp2.Alerts)
+	}
+	if len(paramResp2.Response) < 1 {
+		t.Fatalf("Expected at least one Parameter to exist with name '%s', found: %d", firstParameter2.Name, len(paramResp2.Response))
+	}
+
+	profileID1 := profileResp1.Response[0].ID
+	parameterID1 := paramResp1.Response[0].ID
+	profileID2 := profileResp2.Response[0].ID
```
Comment (ocket8888) [inline traffic_ops/testing/api/v4/profile_parameters_test.go:372]: If the first Profile is the only one you're actually using for the test of assigning multiple Parameters to a Profile, why even bother fetching a second Profile?

File: traffic_ops/traffic_ops_golang/profileparameter/parameterprofilebyname.go [hunk 2]
```diff
@@ -22,9 +22,10 @@ package profileparameter
 import (
 	"database/sql"
 	"errors"
-	"github.com/apache/trafficcontrol/lib/go-util"
 	"net/http"
 
+	"github.com/apache/trafficcontrol/lib/go-util"
+
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/profileparameter/parameterprofilebyname.go:28]: Can you remove this blank line?

Comment (dmohan001c) [reply traffic_ops/traffic_ops_golang/profileparameter/parameterprofilebyname.go]: Removed.

File: traffic_ops/testing/api/v4/profile_parameters_test.go (hunk 1 above)
Comment (dmohan001c) [reply traffic_ops/testing/api/v4/profile_parameters_test.go]: Removed that line.

**Comment by dmohan001c** [comment]
not sure, why the tests are failing here. tests are running without any failure in local.

**Comment by ocket8888** [comment]
There was a timing issue that was recently fixed, you may need to rebase.

**Comment by dmohan001c** [comment]
> There was a timing issue that was recently fixed, you may need to rebase.

Fixed.

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**